### PR TITLE
pacemaker: provide a option to configure migrate-threshold

### PR DESCRIPTION
**Pull request by ilausuch**
Provides a new way to configure pacemaker migrate-threshold option
per frontent. An integer value is passed and it signifies the maximun
numbers of retries before migrate a service to other node. Its value
is 3 by default.

### CODE REVIEW

**Comment by ilausuch** [comment]
[Make migration-threshold configurable for pacemaker cluster](https://trello.com/c/VCyDBWgV/111-make-migration-threshold-configurable-for-pacemaker-cluster)

File: chef/data_bags/crowbar/template-pacemaker.json [hunk 1]
```diff
@@ -38,7 +39,7 @@
           "from": "",
           "server": "",
           "prefix": ""
-	}
+	       }
```
Comment (Itxaka) [inline chef/data_bags/crowbar/template-pacemaker.json]: extra spacing here?

File: chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+def upgrade(ta, td, a, d)
+  # stay compatible with previous behavior
+  a["migration_threshold"] = ta["migration_threshold"]  unless a.key?("migration_threshold")
```
[the remark this answers is not in the document]
Comment (Itxaka) [reply chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb]: agree with the dog, you got an extra space before the `unless`

Comment (vuntz) [inline chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb]: Also the path of the attribute is not correct, you forgot the "crm" part.

File: chef/data_bags/crowbar/template-pacemaker.json [hunk 3]
```diff
@@ -38,7 +39,7 @@
           "from": "",
           "server": "",
           "prefix": ""
-	}
+	      }
```
Comment (vuntz) [inline chef/data_bags/crowbar/template-pacemaker.json]: This spacing change still feels undesired.

File: chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+def upgrade(ta, td, a, d)
+  # stay compatible with previous behavior
+  a["crm"]["migration_threshold"] = ta["crm"]["migration_threshold"] unless a["crm"].key?("migration_threshold")
```
[the remark this answers is not in the document]
Comment (vuntz) [reply chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb]: I guess I'd fix that comment too.

File: chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+def upgrade(ta, td, a, d)
+  # stay compatible with previous behavior
+  a["crm"]["migration_threshold"] = ta["crm"]["migration_threshold"] unless
+  a["crm"].key?("migration_threshold")
```
Comment (vuntz) [inline chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb]: Kind of surprised to not see any warning here, but if the line `bar unless foo` is too long, we generally just go for the form:

```ruby
unless foo
  bar
end
```

Style-wise, what you did here is quite unusual

File: chef/data_bags/crowbar/template-pacemaker.json [hunk 6]
```diff
@@ -39,7 +39,7 @@
           "from": "",
           "server": "",
           "prefix": ""
-	      }
+        }
```
Comment (aspiers) [inline chef/data_bags/crowbar/template-pacemaker.json]: Please can you squash this change into the previous commit?  The motivation for asking is described in https://blog.adamspiers.org/2015/03/24/why-and-how-to-correctly-amend-github-pull-requests/

File: chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb [hunk 7]
```diff
@@ -1,7 +1,8 @@
 def upgrade(ta, td, a, d)
   # stay compatible with previous behavior
-  a["crm"]["migration_threshold"] = ta["crm"]["migration_threshold"] unless
-  a["crm"].key?("migration_threshold")
+  unless a["crm"].key?("migration_threshold")
+    a["crm"]["migration_threshold"] = ta["crm"]["migration_threshold"]
+  end
```
Comment (aspiers) [inline chef/data_bags/crowbar/migrate/pacemaker/201_migration_threshold.rb]: This and the previous commits should all be squashed into one.

**Comment by Itxaka** [comment]
sorry @aspiers I missed that and @ilausuch wanted to ask me about how to squash and rebase commits, so he didnt knew about this. Be assure it will be fixed shortly as we are having a mentoring session about this right now :)

**Comment by aspiers** [comment]
@Itxaka Sounds great, thanks a lot! :-)

**Comment by vuntz** [comment]
@aspiers assuming you're fine with it now

**Pull request merged**